postgres/tests/test_pg_integration.py

```diff
@@ -518,27 +520,68 @@ def test_statement_samples_dbstrict(aggregator, integration_check, dbm_instance,
         conn.close()
 
 
-def test_statement_samples_rate_limits(aggregator, integration_check, dbm_instance, bob_conn):
+def test_statement_samples_main_collection_rate_limit(aggregator, integration_check, dbm_instance, bob_conn):
+    # test the main collection loop rate limit
+    collections_per_second = 10
+    dbm_instance['statement_samples']['collections_per_second'] = collections_per_second
     dbm_instance['statement_samples']['run_sync'] = False
-    # one collection every two seconds
-    dbm_instance['statement_samples']['collections_per_second'] = 0.5
     check = integration_check(dbm_instance)
     check._connect()
-    query = "SELECT city FROM persons WHERE city = 'hello'"
+    check.check(dbm_instance)
+    sleep_time = 1
+    time.sleep(sleep_time)
+    max_collections = int(collections_per_second * sleep_time)
+    check.cancel()
+    metrics = aggregator.metrics("dd.postgres.collect_statement_samples.time")
+    assert max_collections / 2 <= len(metrics) <= max_collections
+
+
+def test_statement_samples_unique_plans_rate_limits(aggregator, integration_check, dbm_instance, bob_conn):
+    # tests rate limiting ingestion of samples per unique (query, plan)
+    cache_max_size = 10
+    dbm_instance['statement_samples']['seen_samples_cache_maxsize'] = cache_max_size
+    # samples_per_hour_per_query set very high so that within this test we will have at most one sample per
```

Comment:
Same comment here as in the [mysql PR](https://github.com/DataDog/integrations-core/pull/9585#discussion_r656564216).

Reply:
Fixed